presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -598,23 +604,63 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName tableName)
+            Map<String, ColumnHandle> columns = columnHandles.stream()
+                    .map(HiveColumnHandle.class::cast)
+                    .filter(not(HiveColumnHandle::isHidden))
+                    .collect(toImmutableMap(HiveColumnHandle::getName, Function.identity()));
+
+            Map<String, Type> columnTypes = columns.entrySet().stream()
+                    .collect(toImmutableMap(Map.Entry::getKey, entry -> getColumnMetadata(session, tableHandle, entry.getValue()).getType()));
+            List<HivePartition> partitions = partitionManager.getPartitions(metastore, tableHandle, constraint, session).getPartitions();
+            return hiveStatisticsProvider.getTableStatistics(session, ((HiveTableHandle) tableHandle).getSchemaTableName(), columns, columnTypes, partitions);
+        }
+
+        verify(!constraint.predicate().isPresent());
+
+        HiveTableLayoutHandle hiveLayoutHandle = (HiveTableLayoutHandle) tableLayoutHandle.get();
+
+        ImmutableSet<ColumnHandle> allColumnHandles = ImmutableSet.<ColumnHandle>builder()
+                .addAll(columnHandles)
+                .addAll(hiveLayoutHandle.getPredicateColumns().values())
+                .build();
+
+        Map<String, ColumnHandle> columns = allColumnHandles.stream()
                 .map(HiveColumnHandle.class::cast)
                 .filter(not(HiveColumnHandle::isHidden))
                 .collect(toImmutableMap(HiveColumnHandle::getName, Function.identity()));
 
-        Map<String, Type> columnTypes = columns.entrySet().stream()
+        Map<ColumnHandle, String> columnNames = columns.entrySet().stream()
+                .collect(toImmutableMap(Map.Entry::getValue, Map.Entry::getKey));
+
+        Map<String, Type> columnTypesByName = columns.entrySet().stream()
                 .collect(toImmutableMap(Map.Entry::getKey, entry -> getColumnMetadata(session, tableHandle, entry.getValue()).getType()));
-        List<HivePartition> partitions = partitionManager.getPartitions(metastore, tableHandle, constraint, session).getPartitions();
-        return hiveStatisticsProvider.getTableStatistics(session, ((HiveTableHandle) tableHandle).getSchemaTableName(), columns, columnTypes, partitions);
+
+        Constraint<ColumnHandle> combinedConstraint = new Constraint<>(constraint.getSummary().intersect(hiveLayoutHandle.getDomainPredicate()
+                .transform(subfield -> isEntireColumn(subfield) ? subfield.getRootName() : null)
+                .transform(columns::get)));
+
+        SubfieldExtractor subfieldExtractor = new SubfieldExtractor(functionResolution, rowExpressionService.getExpressionOptimizer(), session);
+
+        RowExpression domainPredicate = rowExpressionService.getDomainTranslator().toPredicate(
+                hiveLayoutHandle.getDomainPredicate()
+                        .transform(subfield -> subfieldExtractor.toRowExpression(subfield, columnTypesByName.get(subfield.getRootName()))));
```

Comment:
I think `columnTypeByName.get(subfield.getRootName)`, is returning null and causing the failing filter pushdown test.

Reply:
yes

Reply:
This was because there was a column called "$bucket", which didn't exist in columnTypeByName, because were filtering out hidden columns. I undid that filtering out, and now all is well.